flink-python/pyflink/fn_execution/beam/beam_sdk_worker_main.py

```diff
@@ -25,5 +25,44 @@
 
 import apache_beam.runners.worker.sdk_worker_main
 
+
+def custom_logging(logging_func, msg, *args, **kwargs):
+    if msg != '\n':
+        logging_func(msg, *args, **kwargs)
+
+
+class CustomPrint(object):
+    def __init__(self, _print):
+        self._msg_buffer = []
+        self._print = _print
+
+    def print(self, *args, sep=' ', end='\n', file=None):
+        self._msg_buffer.append(sep.join([str(arg) for arg in args]))
+        if end == '\n':
+            self._print(''.join(self._msg_buffer), sep=sep, end=end, file=file)
+            self._msg_buffer.clear()
+        else:
+            self._msg_buffer.append(end)
+
+    def close(self):
+        if self._msg_buffer:
+            self._print(''.join(self._msg_buffer), sep='', end='\n')
+            self._msg_buffer.clear()
+
+
 if __name__ == '__main__':
+    import builtins
+    import logging
+    from functools import partial
+
+    # redirect stdout to logging.info, stderr to logging.error
+    _info = logging.getLogger().info
+    _error = logging.getLogger().error
+    sys.stdout.write = partial(custom_logging, _info)
+    sys.stderr.write = partial(custom_logging, _error)
+
+    _print = print
+    custom_print = CustomPrint(_print)
```

Comment:
```suggestion
    custom_print = CustomPrint(_print=print)
```
Then `_print = print` could be removed